### ISSUE TITLE
Make all built in themes usable by readthedocs

### ISSUE THREAD

**Issue by lendle** [mkdocs/mkdocs#355]
From https://github.com/rtfd/readthedocs.org/issues/978#issuecomment-62574931

> The main reason [themes other than readthedocs can't be used] is because the default mkdocs themes don't have blocks defined, which means we can't override any of the templates. This makes it hard for us to inject some of the page-level context that lets us provide RTD features on top. It's something that we need to get into upstream mkdocs before we can really support it on RTD.

**Comment by d0ugal** [comment]
Okay, that might be fine but we need to know what they need?

**Comment by lendle** [comment]
cc @ericholscher

**Comment by d0ugal** [comment]
If it is just a case of making the block names match the ReadTheDocs theme, that would be easy. However, I think I caused an issue with that recently so it would be good to define this and add some tests for it - A 'Read The Docs' unit (almost integration) test that checks we have what we need would be nice.

**Comment by waylan** [comment]
We added blocks in #947. The same blocks are defined in all of the included themes (including the readthedocs` theme). Of course, third party theme authors have always been free to do as they please. In any event, I'm closing this as "fixed". If any of the existing blocks are not compatible with RTD's needs, then a separate issue can be opened to address that specific need.
